Sources/OpenAI/Public/Service/OpenAIService.swift

```diff
@@ -1157,258 +1173,253 @@ extension OpenAIService {
   /// - Returns: An asynchronous throwing stream of the specified decodable type.
   public func fetchStream<T: Decodable>(
     debugEnabled: Bool,
-    type _: T.Type,
+    type: T.Type,
     with request: URLRequest)
-    async throws -> AsyncThrowingStream<T, Error>
-  {
-    if debugEnabled {
-      printCurlCommand(request)
-    }
+  async throws -> AsyncThrowingStream<T, Error>
+    {
+      if debugEnabled {
+        printCurlCommand(request)
+      }
 
-    let (data, response) = try await session.bytes(
-      for: request,
-      delegate: session.delegate as? URLSessionTaskDelegate)
-    guard let httpResponse = response as? HTTPURLResponse else {
-      throw APIError.requestFailed(description: "invalid response unable to get a valid HTTPURLResponse")
-    }
-    if debugEnabled {
-      printHTTPURLResponse(httpResponse)
-    }
-    guard httpResponse.statusCode == 200 else {
-      var errorMessage = "status code \(httpResponse.statusCode)"
-      do {
-        let data = try await data.reduce(into: Data()) { data, byte in
-          data.append(byte)
-        }
-        let error = try decoder.decode(OpenAIErrorResponse.self, from: data)
-        errorMessage += " \(error.error.message ?? "NO ERROR MESSAGE PROVIDED")"
-      } catch {
-        // If decoding fails, proceed with a general error message
-        errorMessage = "status code \(httpResponse.statusCode)"
+      // Convert URLRequest to HTTPRequest
+      let httpRequest = HTTPRequest(
+        url: request.url!,
+        method: HTTPMethod(rawValue: request.httpMethod ?? "GET") ?? .get,
+        headers: request.allHTTPHeaderFields ?? [:],
+        body: request.httpBody
+      )
```

Comment:
nit: this is repeated several times. Maybe add an initializer in `HTTPRequest` from `URLRequest` ?